neutron/convert.cpp

```diff
@@ -33,12 +34,8 @@ const auto tofToDSpacingPhysicalConstants =
 
 Dataset tofToDSpacing(Dataset &&d) {
   // 1. Compute conversion factor
-  // TODO Is this a good way to store component information?
-  const auto &compPos =
-      d.labels()["component_info"].values<Dataset>()[0]["position"].data();
-  // TODO Need a better mechanism to identify source and sample.
-  const auto &sourcePos = compPos.slice({Dim::Row, 0});
-  const auto &samplePos = compPos.slice({Dim::Row, 1});
+  const auto &sourcePos = source_position(d);
+  const auto &samplePos = sample_position(d);
```

Comment:
Can't you also use the `two_theta` function below where you do `dot(beam, scattered)`?

Reply:
I could, but it feels like a waste. There we actually want `sin(theta)` so I am just using a double-angle identity, instead of computing `acos` then `sin`.